server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -517,17 +518,44 @@ const orderMutations = {
         include: [
           { association: 'paymentMethod' },
           { association: 'fromCollective', attributes: ['id', 'slug'] },
-          { association: 'collective', attributes: ['id', 'slug'] },
+          { association: 'collective', attributes: ['id', 'slug', 'HostCollectiveId'] },
         ],
       });
 
       // -- Some sanity checks to prevent issues --
-      const paymentMethodIds = uniq(orders.map(order => order.PaymentMethodId).filter(Boolean));
+      const isAddedFund = order =>
+        order.paymentMethod?.service === PAYMENT_METHOD_SERVICE.OPENCOLLECTIVE &&
+        order.paymentMethod.type === PAYMENT_METHOD_TYPE.HOST;
+      const paymentMethodIds = uniq(
+        orders
+          .filter(order => !isAddedFund(order))
+          .map(order => order.PaymentMethodId)
+          .filter(Boolean),
+      );
       const ordersIds = orders.map(order => order.id);
+      const addedFundOrders = orders.filter(order => isAddedFund(order));
+      const isUser = fromAccount?.type === types.USER;
+      const addedFundPaymentMethod = fromAccount
+        ? await models.PaymentMethod.findOne({
+            where: {
+              CollectiveId: fromAccount.id,
+              service: 'opencollective',
+              type: 'collective',
+              deletedAt: null,
+            },
+          })
+        : null;
+
       for (const order of orders) {
+        if (fromAccount) {
+          if (fromAccount.HostCollectiveId && fromAccount.HostCollectiveId !== order.collective.HostCollectiveId) {
+            throw new ValidationFailed(`Added Funds cannot be moved to a different host`);
+          }
+        }
```

Comment:
What's that? We're comparing the host of the sender to the host of the Collective?

That doesn't make much sense to me.